test/Exthost/UriTest.re

```diff
@@ -0,0 +1,37 @@
+open TestFramework;
+
+open Oni_Core;
+open Exthost;
+
+let waitForMessage = (~name, f, context) => {
+  context
+  |> Test.waitForMessage(
+       ~name,
+       fun
+       | Msg.MessageService(ShowMessage({message, severity, _})) =>
+         f(severity, message)
+       | _ => false,
+     );
+};
+
+describe("UriTest", ({test, _}) => {
+  test("revive uri", _ => {
+    Test.startWithExtensions(["oni-uri"])
+    |> Test.waitForExtensionActivation("oni-uri")
+    |> Test.withClient(
+         Request.Commands.executeContributedCommand(
```

Comment:
Cool, thanks for adding the test! I wasn't sure where we were hitting this, but it is clear now.

Reply:
If you want more details of the code path:

https://github.com/onivim/vscode-exthost/blob/923c38b016c87a205957456e13c62f8dfd3bdc62/src/vs/workbench/api/common/extHostCommands.ts#L189

https://github.com/onivim/vscode-exthost/blob/923c38b016c87a205957456e13c62f8dfd3bdc62/src/vs/workbench/api/common/extHostCommands.ts#L53

https://github.com/onivim/vscode-exthost/blob/923c38b016c87a205957456e13c62f8dfd3bdc62/src/vs/base/common/marshalling.ts#L42-L45

https://github.com/onivim/vscode-exthost/blob/923c38b016c87a205957456e13c62f8dfd3bdc62/src/vs/base/common/uri.ts#L384-L395

So you see it's the marshaller that uses `$mid` for disambiguation. `URI.revive` doesn't even check it, so if something just assumes it's a URI it would have worked fine.